round4/team-9.md

```diff
@@ -0,0 +1,132 @@
+
+## Technical Details
+
+### User Interface
+
+Users will interact with a sidebar.
+
+### API, Libraries, and Frameworks
+
+**Chrome Tabs API:** We plan to use this to actually interact with browser tabs, retrieving them and grouping them together.
+
+**OpenAI API:** We plan to use this to figure out automated suggestions for groupings of tabs that users frequently visit.
+
+**Selenium:** We plan to use this library to figure out how to scrape login data from websites and log in for the user when opening tabs for sites that necessitate logging in (ex. Canvas).
+
+**React:** We plan to use React for the front-end visualization of our application.
+
+**Node.js:** We plan to use Node as the backend for our application (for data processing, user authentication, and integration with other APIs).
+
+**Google Analytics:** We plan to use Google Analytics to examine the user's time-spending information and generate insights through them.
+
+### Data Storage
+
+Each tab group should have its own model to store information such as:
+- _Group name:_ A descriptive name for the tab group.
+- _Group color:_ If color-coding is supported, store the chosen color.
+- _Tabs:_ An array or list of tab objects representing the tabs contained within the group.
+- _Time spent in tab group:_ For remainders, insights, etc.
+- _Optional properties:_ Any additional properties related to the tab group, such as creation date, pinned status, etc.
+
+Each tab within a tab group should be represented by a model containing details like:
+- _Tab URL:_ The URL of the tab.
+- _Tab title:_ The title of the webpage.
+
+## Project Management
+
+### Collaboration and Task Allocation
+
+- **Leader:** Eshann (OpenAI API Integration)
+- **Manager:** Gui (Tab Grouping)
+- **Remaining Team Members:** Anirudh (UI/UX), Kofi (Google Analytics)
+
+### Risks and Mitigation
+
+**Complexity of Integration:** Integrating multiple APIs (Chrome Tabs API, OpenAI API, etc.) and frameworks (React, Node.js) may lead to technical challenges and delays.
+
+**Mitigation Strategy:** We will try to mitigate this by breaking down our development tasks into smaller, manageable chunks, as well as conducting thorough research of each API/library before implementing anything involving it. We also plan to continually seek guidance from Professor Lumbroso and the TAs for any technical issues encountered.
+
+**API Limitations or Changes:** The APIs used in the extension may have limitations or undergo changes that could impact functionality or compatibility.
+
+**Mitigation Strategy:** We will do our best to Implement error handling and fallback mechanisms to handle API limitations or failures.
+**Security Vulnerabilities:** Inadequate handling of user data or authentication mechanisms may expose our extension to security risks such as data breaches or unauthorized access attempts.
+
+**Mitigation Strategy:** We will do our best to implement robust authentication and authorization mechanisms to protect user data.
```

Comment:
Suggestion to focus on Penn student related websites (i.e., anything behind a Penn single sign on).